cc3d/CompuCellSetup/CC3DPy.py

```diff
@@ -127,6 +128,85 @@ def visualization():
         frame.launch()
         return frame
 
+    @staticmethod
+    def get_screenshot_data() -> dict:
+        """Gets a copy of the current screenshot data"""
+
+        sm = CompuCellSetup.persistent_globals.screenshot_manager
+        if sm is None:
+            raise RuntimeError('Screenshot manager not available at this time')
+
+        return sm.generate_screenshot_description_data()
+
+    @staticmethod
+    def set_screenshot_data(_data: dict):
+        """Sets current screenshot data"""
+        sm = CompuCellSetup.persistent_globals.screenshot_manager
+        if sm is None:
+            raise RuntimeError('Screenshot manager not available at this time')
+
+        sm.read_screenshot_description_data(_data)
+
+    @staticmethod
+    def save_screenshot_data(_fp: str = None) -> str:
+        """
+        Saves screenshot data if available
+
+        :param _fp: absolute file path of saved data; default is standard location if available
+        :return: file path if saved, otherwise None
+        """
+
+        sm = CompuCellSetup.persistent_globals.screenshot_manager
+        if sm is None:
+            raise RuntimeError('Screenshot manager not available at this time')
+
+        if _fp is None:
+            _fp = sm.get_screenshot_filename()
+        if _fp is None:
+            raise RuntimeError('Screenshot file name could not be deduced')
+
+        sm.write_screenshot_description_file(_fp)
+        return _fp
+
+    @staticmethod
+    def load_screenshot_data(_fp: str):
+        """
+        Loads screenshot data from file
+
+        :param _fp: absolute path of saved data
+        :return: true is loaded
+        """
+
+        if not os.path.isfile(_fp):
+            raise FileNotFoundError('Screenshot data file does not exist at path', _fp)
+
+        sm = CompuCellSetup.persistent_globals.screenshot_manager
+        if sm is None:
+            raise RuntimeError('Screenshot manager not available at this time')
```

Comment:
nit: Since these lines are used 4 times, it could be extracted into a private get method that throws the error.